gprofiler/log.py

```diff
@@ -89,101 +41,49 @@ class RemoteLogsHandler(logging.Handler):
-
-    def init_api_client(self, api_client: "APIClient") -> None:
-        self._api_client = api_client
+    def __init__(self, server_address: str, auth_token: str, service_name: str) -> None:
+        self._service_name = service_name
+        url = urlparse(server_address)
+        super().__init__(
+            Sender(application_name="gprofiler", auth_token=auth_token, scheme=url.scheme, server_address=url.netloc)
+        )
 
     def emit(self, record: LogRecord) -> None:
-        extra_kwargs = record.gprofiler_adapter_extra  # type: ignore
-        if extra_kwargs.pop(NO_SERVER_LOG_KEY, False):
+        if self.get_extra_fields(record).pop(NO_SERVER_LOG_KEY, False):
             return
 
-        self._logs.append(self._make_dict_record(record))
-        # truncate logs to last N entries
-        if len(self._logs) > self.MAX_BUFFERED_RECORDS:
-            self._truncated = True
-            self._logs[: -self.MAX_BUFFERED_RECORDS] = []
-
-    def _make_dict_record(self, record: LogRecord) -> Dict[str, Any]:
-        formatted_timestamp = datetime.datetime.utcfromtimestamp(record.created).isoformat()
-        extra = record.gprofiler_adapter_extra  # type: ignore
-
-        run_id = extra.pop(RUN_ID_KEY, None)
-        if run_id is None:
-            self._logger.error("state.run_id is not defined! probably a bug!")
-            run_id = ""
-
-        cycle_id = extra.pop(CYCLE_ID_KEY, "")
-
-        # We don't want to serialize a JSON inside JSON but either don't want to fail record because of extra
-        # serialization, so we test if the extra can be serialized and have a fail-safe.
-        try:
-            json.dumps(extra)
-        except TypeError:
-            self._logger.exception(
-                f"Can't serialize extra (extra={extra!r}), sending empty extra", bad_extra=repr(extra)
-            )
-            extra = {}
-
-        assert self.formatter is not None
-        if record.exc_info:
-            # Use cached exc_text if available.
-            exception_traceback = (
-                record.exc_text if record.exc_text else self.formatter.formatException(record.exc_info)
-            )
-        else:
-            exception_traceback = ""
-
-        extra = extra if not extra.pop(NO_SERVER_EXTRA_KEY, False) else {}
-
-        return {
-            "message": record.message,
-            "level": record.levelname,
-            "timestamp": formatted_timestamp,
-            "extra": extra,
-            "logger_name": record.name,
-            "exception": exception_traceback,
-            RUN_ID_KEY: run_id,
-            CYCLE_ID_KEY: cycle_id,
+        if self.get_extra_fields(record).pop(NO_SERVER_EXTRA_KEY, False):
+            record.extra = {}
+
+        return super().emit(record)
+
+    def get_metadata(self) -> Dict[str, str]:
+        from gprofiler.metadata.system_metadata import get_hostname_or_none
+
+        state = get_state()
+        hostname = get_hostname_or_none()
+
+        metadata = {
```

Comment:
I kept the same set of metadatas as we had in the previous version. Will add more on demand.